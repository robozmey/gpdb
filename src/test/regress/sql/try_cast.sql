create schema trycast;
set search_path = trycast;

-- no way to cast
select try_cast(12 as date);

-- typecast via I/O
select try_cast('11/11/20111' as int);
select try_cast('111d' as int);

-- typecast from pg_proc
select try_cast('112344466343'::int8 as int4);

-- arrays
select try_cast('{1, 2, 31111111111}'::int8[] as int4[]); -- pg_proc
select try_cast('{1, 2, lol}'::text[] as int4[]); -- via I/O
select try_cast('{1, 2, 31111111111}' as int4[]); -- 

reset search_path;