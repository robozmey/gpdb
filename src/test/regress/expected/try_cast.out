create schema trycast;
set search_path = trycast;

-- no way to cast
select try_cast(12 as date);
 date 
------
 
(1 row)

-- typecast via I/O
select try_cast('11/11/20111' as int);
 int4 
------
     
(1 row)
select try_cast('111d' as int);
 int4 
------
     
(1 row)

-- typecast from pg_proc
select try_cast('112344466343'::int8 as int4);
 int4 
------
     
(1 row)

-- arrays
select try_cast('{1, 2, 31111111111}'::int8[] as int4[]); -- pg_proc
 int4 
------
 {1,2,NULL}
(1 row)
select try_cast('{1, 2, lol}'::text[] as int4[]); -- via I/O
 int4 
------
 {1,2,NULL}
(1 row)
select try_cast('{1, 2, 31111111111}' as int4[]); -- 
 int4 
------
 {1,2,NULL}
(1 row)

reset search_path;